elastic/kibana
# [Ingest Manager] Remove fields from index pattern during package uninstall

## Summary

This PR addresses this issue: https://github.com/elastic/kibana/issues/80031

The `installIndexPatterns()` function relies on the saved objects being absent for removing the fields from a package during uninstall. Since the saved object is still present when the function is called it will collection the same fields as before instead of removing them fields for the package being uninstalled.

## Testing

1. Start a fresh kibana and elasticsearch
2. Navigate to the ingest manager page to trigger `/setup`
3. Install the crowdstrike `0.1.1` package

```
curl --location --request POST 'http://elastic:changeme@localhost:5601/api/fleet/epm/packages/crowdstrike-0.1.1' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--header 'kbn-xsrf: xxx' \
--data-raw ''
```
5. Navigate back to the index pattern management page and observe that the crowdstrike fields exist
![image](https://user-images.githubusercontent.com/56361221/95490242-f6174880-0965-11eb-8e11-76bfbff4b8b7.png)



6. Uninstall the crowdstrike package

```
curl --location --request DELETE 'http://elastic:changeme@localhost:5601/api/fleet/epm/packages/-0.1.1' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--header 'kbn-xsrf: xxx' \
--data-raw ''
```

6. Navigate back to the index pattern management page and observe that the crowdstrike fields are removed.

![image](https://user-images.githubusercontent.com/56361221/95514180-83b85f80-0989-11eb-8f18-14bbb32ff197.png)
